crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -169,10 +169,8 @@ impl Diagnostic {
             return None;
         }
 
-        if let Some(newest) = self.history.back() {
-            if let Some(oldest) = self.history.front() {
-                return Some(newest.time.duration_since(oldest.time));
-            }
+        if let (Some(newest), Some(oldest)) = (self.history.back(), self.history.front()) {
```

Comment:
```rust
    pub fn duration(&self) -> Option<Duration> { 
        if self.history.len() < 2 { 
            return None;
        } 

        if let (Some(newest), Some(oldest)) = (self.history.back(), self.history.front()) { 
            Some(newest.time.duration_since(oldest.time))
        } else { 
            None
        } 
    }
```

Reply:
A bit cleaner IMO, no big deal either way.

Reply:
I think the function body can be replaced with
```rust
self.history.back().zip(self.history.front()).map(|(newest, oldest)| newest.time.duration_since(oldest.time))
```
The wonders (to some, horrors) of monadic composition!
I think the whole function body can be replaced with that *if* it's fine to have `Some(Duration(0.0))` (if `self.history.len()` is 1)